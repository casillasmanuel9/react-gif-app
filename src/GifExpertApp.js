import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    //const categories = ['One Punch','Samuray X', 'Dragon Ball']
    const [categories, setCategories] = useState(['One Punch']);

    /*const handleAdd = () => {
        //setCategories([...categories, 'HunterXhunter'] );
        setCategories( cats => [...cats,'HunterXhunter']);
    }*/

    return (
        <>
            <h2>GifExpertAPp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>

            <button> Agregar </button>

            <ol>
                {categories.map(category=> (
                    <GifGrid key={category} category={category}/>
                ))}
            </ol>
        </>
    )
}
