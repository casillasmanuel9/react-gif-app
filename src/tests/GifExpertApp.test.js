import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas en el componente GifExpertApp', () => {
    test('Debe de hacer match con el snaptshoot', () => {
        const wraper = shallow(<GifExpertApp />);
        expect(wraper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de catergorias', () => {
        const categoryes =  ['Halo','Xbox'];
        const wraper = shallow(<GifExpertApp defaultCategories={categoryes}/>);
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('GifGrid').length).toBe(categoryes.length);
    });
});
