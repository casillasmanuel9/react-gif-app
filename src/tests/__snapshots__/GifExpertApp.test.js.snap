// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifExpertApp Debe de hacer match con el snaptshoot 1`] = `
<Fragment>
  <h2>
    GifExpertAPp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button>
     Agregar 
  </button>
  <ol />
</Fragment>
`;

exports[`Pruebas en el componente GifExpertApp Debe de mostrar una lista de catergorias 1`] = `
<Fragment>
  <h2>
    GifExpertAPp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button>
     Agregar 
  </button>
  <ol>
    <GifGrid
      category="Halo"
      key="Halo"
    />
    <GifGrid
      category="Xbox"
      key="Xbox"
    />
  </ol>
</Fragment>
`;
